### ISSUE TITLE
Dimensional_oM : Rename IElement -> IElementXD

### ISSUE THREAD

**Issue by al-fisher** [BHoM/BHoM#743]
Following the formalisation of the Spatial Elements, the Spatial Engine and further building out of Matter Engine etc. useful now to clarify the naming of the current `IElement`

The `IElement` as currently implemented is the base interface for only the `IElement0D`, `IElement1D` & `IElement2D`. 
With now other types of Element - `IElementM` etc. is proposed to rename `IElement` to clarify it is only a parent to Spatial Elements.

**Propose rename to IElementXD**

Alternative discussed has been IElementL - in reference to the Length dimension. However think clearer to name IElementXD.

Hope makes sense
@IsakNaslundBh @pawelbaran @kThorsager (and anyone else) FYI can you perhaps confirm happy with new name? 

Actioning in 3.2 milestone after current work is settled

**Comment by pawelbaran** [comment]
I will not make it easier 😉 what about `IElementS`, to be consistent with `Spatial`?

**Comment by IsakNaslundBh** [comment]
My official vote goes to: 🥁 🥁 🥁 

`IElementXD`

**Comment by kThorsager** [comment]
Think we're both unofficially referring to them as `IElementXD` from time to time and keeping the `D` part somewhere in there is preferable in my opinion if we're not renaming `IElement0D` etc as well

**Comment by pawelbaran** [comment]
xD

**Comment by al-fisher** [comment]
> xD

@pawelbaran does your comment above mean `IElementXD` gets your vote? 😄

**Comment by pawelbaran** [comment]
No, I just could not resist putting a smiley face 😃 Jokes aside, I do not have a strong opinion on that subject, my intuitive idea was to name it `IElementS` to couple with Spatial, but I will be happy with any option
